Assets/scripts/Singleton.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+public class Singleton<T> : MonoBehaviour where T : MonoBehaviour
+{
+
+
+    private static T instance;
+
+
+    public static T Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                instance = FindObjectOfType<T>();
+
+
+                DontDestroyOnLoad(instance);
+            }
+            else if (instance != FindObjectOfType<T>())
+            {
+                Destroy(instance);
```

Comment:
when instance != null => means that this component is a duplicate and needs to be deleted. so don't destroy instance, instead destroy this